sklearn/utils/tests/test_param_validation.py

```diff
@@ -318,19 +314,15 @@ def test_generate_invalid_param_val_2_intervals(integer_interval, real_interval)
         [_RandomStates()],
         [_SparseMatrices()],
         [_Booleans()],
-        [Interval(Real, None, None, closed="both")],
-        [
-            Interval(Integral, 0, None, closed="left"),
-            Interval(Real, None, 0, closed="neither"),
-        ],
+        [Interval(Integral, None, None, closed="neither")],
     ],
 )
 def test_generate_invalid_param_val_all_valid(constraints):
     """Check that the function raises NotImplementedError when there's no invalid value
     for the constraint.
     """
     with pytest.raises(NotImplementedError):
-        generate_invalid_param_val(constraints[0], constraints=constraints)
+        generate_invalid_param_val(constraints[0])
```

Comment:
with the API change to `generate_invalid_param_val`, this test can be updated to only accept single constraints?

```python
def test_generate_invalid_param_val_all_valid(constraint):
    with pytest.raises(NotImplementedError):
        generate_invalid_param_val(constraint)
```

Reply:
indeed thanks